mingw cross-compilation failed

<!-- This issue tracker is only for technical issues related to Bitcoin Core.

General bitcoin questions and/or support requests are best directed to the Bitcoin StackExchange at https://bitcoin.stackexchange.com.

For reporting security issues, please read instructions at https://bitcoincore.org/en/contact/.

If the node is "stuck" during sync or giving "block checksum mismatch" errors, please ensure your hardware is stable by running memtest and observe CPU temperature with a load-test tool such as linpack before creating an issue! -->

<!-- Describe the issue -->
<!--- What behavior did you expect? -->

<!--- What was the actual behavior (provide screenshots if the issue is GUI-related)? -->

<!--- How reliably can you reproduce the issue, what are the steps to do so? -->

<!-- What version of Bitcoin Core are you using, where did you get it (website, self-compiled, etc)? -->

<!-- What type of machine are you observing the error on (OS/CPU and disk type)? -->

<!-- For the GUI-related issue on Linux provide names and versions of a distro, a desktop environment and a graphical shell (if relevant). -->

<!-- Any extra information that might be useful in the debugging process. -->
<!--- This is normally the contents of a `debug.log` or `config.log` file. Raw text or a link to a pastebin type site are preferred. -->

Build lava windows failed following the instruction [https://github.com/lavaio/lava/blob/master/doc/build-windows.md#building-for-64-bit-windows](url)

```shell
  CXX      libbitcoin_common_a-poc.o
In file included from /usr/lib/gcc/x86_64-w64-mingw32/7.3-posix/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-w64-mingw32/7.3-posix/include/x86intrin.h:48,
                 from /usr/share/mingw-w64/include/winnt.h:1519,
                 from /usr/share/mingw-w64/include/minwindef.h:163,
                 from /usr/share/mingw-w64/include/windef.h:8,
                 from /usr/share/mingw-w64/include/windows.h:69,
                 from /usr/share/mingw-w64/include/winsock2.h:23,
                 from ./compat.h:25,
                 from ./util/system.h:18,
                 from ./dbwrapper.h:12,
                 from ./ticket.h:8,
                 from ./primitives/transaction.h:14,
                 from ./primitives/block.h:9,
                 from ./chain.h:11,
                 from poc.cpp:2:
/usr/lib/gcc/x86_64-w64-mingw32/7.3-posix/include/avxintrin.h: In function ‘std::vector<unsigned char> genNonceChunk(uint64_t, uint64_t)’:
/usr/lib/gcc/x86_64-w64-mingw32/7.3-posix/include/avxintrin.h:607:1: error: inlining failed in call to always_inline ‘void _mm256_zeroupper()’: target specific option mismatch
 _mm256_zeroupper (void)
 ^~~~~~~~~~~~~~~~
poc.cpp:15:47: note: called from here
 #define MMZEROUPPER()         _mm256_zeroupper()
                               ~~~~~~~~~~~~~~~~^~
poc.cpp:59:5: note: in expansion of macro ‘MMZEROUPPER’
     MMZEROUPPER();
     ^
In file included from /usr/lib/gcc/x86_64-w64-mingw32/7.3-posix/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-w64-mingw32/7.3-posix/include/x86intrin.h:48,
                 from /usr/share/mingw-w64/include/winnt.h:1519,
                 from /usr/share/mingw-w64/include/minwindef.h:163,
                 from /usr/share/mingw-w64/include/windef.h:8,
                 from /usr/share/mingw-w64/include/windows.h:69,
                 from /usr/share/mingw-w64/include/winsock2.h:23,
                 from ./compat.h:25,
                 from ./util/system.h:18,
                 from ./dbwrapper.h:12,
                 from ./ticket.h:8,
                 from ./primitives/transaction.h:14,
                 from ./primitives/block.h:9,
                 from ./chain.h:11,
                 from poc.cpp:2:
/usr/lib/gcc/x86_64-w64-mingw32/7.3-posix/include/avxintrin.h:607:1: error: inlining failed in call to always_inline ‘void _mm256_zeroupper()’: target specific option mismatch
 _mm256_zeroupper (void)
 ^~~~~~~~~~~~~~~~
poc.cpp:15:47: note: called from here
 #define MMZEROUPPER()         _mm256_zeroupper()
                               ~~~~~~~~~~~~~~~~^~
poc.cpp:59:5: note: in expansion of macro ‘MMZEROUPPER’
     MMZEROUPPER();
     ^
Makefile:7079: recipe for target 'libbitcoin_common_a-poc.o' failed
make[2]: *** [libbitcoin_common_a-poc.o] Error 1
make[2]: Leaving directory '/home/fun/lava/src'
Makefile:12589: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/fun/lava/src'
Makefile:775: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

--- a/configure.ac
+++ b/configure.ac
@@ -1424,6 +1424,12 @@ AM_CONDITIONAL([ENABLE_AVX2],[test x$enable_avx2 = xyes])
 AM_CONDITIONAL([ENABLE_SHANI],[test x$enable_shani = xyes])
 AM_CONDITIONAL([USE_ASM],[test x$use_asm = xyes])
 
+case ${host} in
+   *mingw*)
+     AC_DEFINE(WITHOUT_ASM, 1, [Define if use mingw toolchain to disalble use shabal.asm])
+     AM_CONDITIONAL([WITHOUT_ASM],[test 1])
+   ;;
+esac
 AC_DEFINE(CLIENT_VERSION_MAJOR, _CLIENT_VERSION_MAJOR, [Major version])
 AC_DEFINE(CLIENT_VERSION_MINOR, _CLIENT_VERSION_MINOR, [Minor version])
 AC_DEFINE(CLIENT_VERSION_REVISION, _CLIENT_VERSION_REVISION, [Build revision])

--- a/configure.ac
+++ b/configure.ac
@@ -1423,11 +1423,11 @@ AM_CONDITIONAL([ENABLE_SSE41],[test x$enable_sse41 = xyes])
 AM_CONDITIONAL([ENABLE_AVX2],[test x$enable_avx2 = xyes])
 AM_CONDITIONAL([ENABLE_SHANI],[test x$enable_shani = xyes])
 AM_CONDITIONAL([USE_ASM],[test x$use_asm = xyes])
+AM_CONDITIONAL([WITHOUT_ASM],[test $host = *mingw*])
 
 case ${host} in
    *mingw*)
      AC_DEFINE(WITHOUT_ASM, 1, [Define if use mingw toolchain to disalble use shabal.asm])
-     AM_CONDITIONAL([WITHOUT_ASM],[test 1])
    ;;
 esac
 AC_DEFINE(CLIENT_VERSION_MAJOR, _CLIENT_VERSION_MAJOR, [Major version])

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -422,6 +422,9 @@ libbitcoin_consensus_a_SOURCES = \
 # common: shared between bitcoind, and bitcoin-qt and non-server tools
 libbitcoin_common_a_CPPFLAGS = $(AM_CPPFLAGS) $(BITCOIN_INCLUDES)
 libbitcoin_common_a_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
+if WITHOUT_ASM
+libbitcoin_common_a_CXXFLAGS += -DWITHOUT_ASM
+endif
 libbitcoin_common_a_SOURCES = \
   base58.cpp \
   bech32.cpp \

--- a/src/poc.cpp
+++ b/src/poc.cpp
@@ -5,7 +5,7 @@
 
 using namespace std;
 
-#if defined(WIN32) && !defined(WHITOUT_ASM)
+#if defined(WIN32) && !defined(WITHOUT_ASM)
 #include <immintrin.h>
 #include <shabal/shabal.h>
 #define CONTEXT               shabal_context

